# Patch adapted from
#
#    https://aur.archlinux.org/cgit/aur.git/tree/0001-only-build-the-module-in-dkms.conf.patch?h=zfs-dkms
#
# rebased for zfs-2.1.5.
#
# Avoids recompiling ZFS userland utilities with DKMS rebuilds
From b4a2c0b184c9c9599421b15a430fb88deb5dbd17 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Sun, 28 Oct 2018 15:01:58 -0400
Subject: [PATCH] only build the module in dkms.conf

---
 scripts/dkms.mkconf | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/scripts/dkms.mkconf b/scripts/dkms.mkconf
--- a/scripts/dkms.mkconf
+++ b/scripts/dkms.mkconf
@@ -28,14 +28,7 @@
 PRE_BUILD="configure
   --prefix=/usr
   --with-config=kernel
-  --with-linux=\$(
-    if [ -e "\${kernel_source_dir/%build/source}" ]
-    then
-      echo "\${kernel_source_dir/%build/source}"
-    else
-      echo "\${kernel_source_dir}"
-    fi
-  )
+  --with-linux="\${kernel_source_dir}"
   --with-linux-obj="\${kernel_source_dir}"
   \$(
     [[ -n \"\${ICP_ROOT}\" ]] && \\
@@ -68,7 +61,7 @@
   -t \${dkms_tree}
 "
 AUTOINSTALL="yes"
-MAKE[0]="make"
+MAKE[0]="make -C module/"
 STRIP[0]="\$(
   [[ -r \${PACKAGE_CONFIG} ]] \\
   && source \${PACKAGE_CONFIG} \\
